kratos/processes/find_intersected_geometrical_objects_process.cpp

```diff
@@ -8,248 +8,476 @@
+{
+    return mrModelPartIntersecting;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+OctreeBinary<OctreeBinaryCell<typename FindIntersectedGeometricalObjectsProcess::ConfigurationType>>* FindIntersectedGeometricalObjectsProcess::GetOctreePointer()
+{
+    return& mOctree;
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindIntersectedGeometricalObjectsProcess::Clear()
+{
+    mIntersectedObjects.clear();
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindIntersectedGeometricalObjectsProcess::Execute()
+{
+    // Calling initialize first (initialize Octree)
+    ExecuteInitialize();
+
+    OtreeCellVectorType leaves;
+
+    // Iterate over elements
+    auto& r_elements_array = mrModelPartIntersected.Elements();
+    const SizeType number_of_elements = r_elements_array.size();
+
+    const auto it_elements_begin = r_elements_array.begin();
+
+    #pragma omp parallel for private(leaves)
+    for (int i = 0; i < static_cast<int>(number_of_elements); i++) {
+        auto it_elem = it_elements_begin + i;
+        leaves.clear();
+        IdentifyNearEntitiesAndCheckEntityForIntersection(*(it_elem.base()), leaves);
     }
 
-    /// Print information about this object.
-    void FindIntersectedGeometricalObjectsProcess::PrintInfo(std::ostream& rOStream) const {
-        rOStream << Info();
+    // Iterate over conditions
+    auto& r_conditions_array = mrModelPartIntersected.Conditions();
+    const SizeType number_of_conditions = r_conditions_array.size();
+
+    const auto it_conditions_begin = r_conditions_array.begin();
+
+    #pragma omp parallel for private(leaves)
+    for (int i = 0; i < static_cast<int>(number_of_conditions); i++) {
+        auto it_cond = it_conditions_begin + i;
+        leaves.clear();
+        IdentifyNearEntitiesAndCheckEntityForIntersection(*(it_cond.base()), leaves);
     }
+}
 
-    /// Print object's data.
-    void FindIntersectedGeometricalObjectsProcess::PrintData(std::ostream& rOStream) const {
+/***********************************************************************************/
+/***********************************************************************************/
+
+void FindIntersectedGeometricalObjectsProcess::ExecuteInitialize()
+{
+    GenerateOctree();
+}
 
+/***********************************************************************************/
+/***********************************************************************************/
+
+std::size_t FindIntersectedGeometricalObjectsProcess::WorkingSpaceDimension()
+{
+    auto& r_elements_array = mrModelPartIntersected.Elements();
```

Comment:
```suggestion
    const auto& r_elements_array = mrModelPartIntersected.Elements();
```
(just suggesting if you didn't try it)